### PR TITLE
Column pooler2 update

### PR DESCRIPTION
**Pull request by breznak**
updates to CP branch
- merges master (makes this PR a bit messy) 
- uses upstream MNIST experiment
- small fixes to CP
- all compiles & runs
  - but I'm unlucky, MNIST score is again ~20% :thinking:

### CODE REVIEW

**Comment by breznak** [comment]
@ctrl-z-9000-times

File: src/nupic/algorithms/ColumnPooler.cpp [hunk 1]
```diff
@@ -42,6 +42,10 @@
 #include <nupic/math/Math.hpp>
 #include <nupic/math/Topology.hpp>
 
+namespace nupic {
+namespace algorithms {
+namespace column_pooler {
```
Comment (breznak) [inline src/nupic/algorithms/ColumnPooler.cpp]: added ns

File: src/nupic/algorithms/ColumnPooler.cpp [hunk 2]
```diff
@@ -61,7 +65,7 @@ class ColumnPooler // : public Serializable
   vector<UInt> distalInputDimensions_;
   vector<UInt> inhibitionDimensions_;
   vector<UInt> cellDimensions_;
-  UInt         cellsPerInhbitionArea_;
+  UInt         cellsPerInhibitionArea_;
```
Comment (breznak) [inline src/nupic/algorithms/ColumnPooler.cpp]: confusing name/typo

File: src/nupic/algorithms/ColumnPooler.cpp [hunk 3]
```diff
@@ -112,11 +116,65 @@ class ColumnPooler // : public Serializable
 
   Connections distalConnections;
 
+
+  ColumnPooler() {}; //default constructor, must call initialize to setup properly
```
Comment (breznak) [inline src/nupic/algorithms/ColumnPooler.cpp]: default constructor & initialize()

File: src/nupic/algorithms/ColumnPooler.cpp [hunk 4]
```diff
@@ -400,6 +458,7 @@ class ColumnPooler // : public Serializable
     for(const auto &cell : active.getFlatSparse())
     {
       // Adapt Proximal Segments
+      NTA_CHECK(cell < proximalMaxSegment_.size()) << "cell oob! " << cell << " < " << proximalMaxSegment_.size();
```
Comment (breznak) [inline src/nupic/algorithms/ColumnPooler.cpp]: not sure, but sometimes I may crash here

**Comment by ctrl-z-9000-times** [comment]
Thanks for working on this @breznak!
I have a few changes on my own branch which I want to merge with your changes when I get the chance, and then I will merge this into the main development branch `columnPooler2`

**Comment by breznak** [comment]
> I have a few changes on my own branch which I want to merge with your changes when I get the chance, and then I will merge this into the main development branch columnPooler2 

thanks! This was working ok, but some merge conglict broke between me merging your 2nd push to cp2 branch, and merging master.. I'll be trying to fix that.

**Comment by breznak** [comment]
Thanks for merging this in the main devel branch. Closing.

**Pull request closed without merging**